src/restore.c

```diff
@@ -388,23 +448,17 @@ restore_main(int32_t argc, char **argv)
 		fclose(mach_fd);
 	}
 
-	restore_status_destroy(&status);
+	if (res == EXIT_FAILURE) {
+		restore_status_destroy(status);
+		cf_free(status);
+		status = RUN_RESTORE_FAILURE;
+	}
 
 cleanup1:
-	restore_config_destroy(&conf);
 
-	file_proxy_cloud_shutdown();
-
-	ver("Exiting with status code %d", res);
-
-	return res;
```

Comment:
I assume the caller now handles this stuff. It's a bit difficult to read the diff because it is mostly code moved around

Reply:
If the user of the shared lib uses `restore_run` it handles the cloud shutdown but the caller needs to destroy the conf. It's outlined in the docstring for `restore_run`